apps/web/src/components/Common/TopicSelector.tsx

```diff
@@ -0,0 +1,130 @@
+import { useFormatMessage } from "@litespace/ui/hooks/intl";
+import React, { useCallback, useMemo, useState } from "react";
+import TopicChanger from "@/components/Common/TopicChanger";
+import { useTopics, useUserTopics } from "@litespace/headless/topic";
+import { useUpdateUserTopics } from "@litespace/headless/user";
+import { TopicSelectionDialog } from "@litespace/ui/TopicSelectionDialog";
+import { useToast } from "@litespace/ui/Toast";
+import { useInvalidateQuery } from "@litespace/headless/query";
+import { QueryKey } from "@litespace/headless/constants";
+import { useUserContext } from "@litespace/headless/context/user";
+import { IUser } from "@litespace/types";
+
+export const TopicSelector: React.FC = () => {
+  const [showTopicsDialog, setShowTopoicsDialog] = useState<boolean>(false);
```

Comment:
There is no other dialog so it is safe to write like this.

```suggestion
  const [showDialog, setShowDialog] = useState<boolean>(false);
```